File: dagger/src/main/java/com/laioffer/youtube/demo2_构造器注入依赖/Demo2Activity.java
package com.laioffer.youtube.demo2_构造器注入依赖;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.laioffer.R;

/**
 * 类的描述:
 * Created by 春夏秋冬在中南 on 2023/8/13 13:33
 */
public class Demo2Activity extends AppCompatActivity {

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.empty_layout);

    Engine engine = new Engine();
    Wheels wheels = new Wheels();

    Car car = new Car(engine, wheels);
    car.drive();
  }
}

File: jetpack/src/main/java/com/laioffer/lesson1_lifecycle/demo2_mvp/Demo2Activity.kt
package com.laioffer.lesson1_lifecycle.demo2_mvp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.laioffer.R

/**
 * 类描述: 版本2: MVP，通过P层 监听生命周期函数
 * created by 春夏秋冬在中南 on 2023/11/28 23:31
 */
class Demo2Activity : AppCompatActivity() {

  private var mMyPresenter: MyPresenter? = null

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.empty_layout)

    mMyPresenter = MyPresenter()
  }

  override fun onResume() {
    super.onResume()
    mMyPresenter?.onResume()
  }

  override fun onPause() {
    super.onPause()
    mMyPresenter?.onPause()
  }
}